pkg/types/vsphere/platform.go

```diff
@@ -47,4 +47,7 @@ type Platform struct {
 
 	// Network specifies the name of the network to be used by the cluster.
 	Network string `json:"network,omitempty"`
+
+	// Thin specifies if thin disks should be use instead of thick
+	Thin bool `json:"thin,omitempty"`
```

Comment:
Can we make this an enum rather than a boolean to support the unlikely case of more options other than thick and thin in the future?
```suggestion
	DiskProvisioning DiskProvisioning `json:"diskProvisioning,omitempty"`
```

```
// +kubebuilder:validation:Enum="";Thin;Thick
type DiskProvisioning string

const (
	ThinDiskProvisioning DiskProvisioning = "Thin"

	ThickDiskProvisioning DiskProvisioning = "Thick"
)
```

Reply:
@ayesha54 check out: 
https://github.com/vmware/govmomi/blob/master/vim25/types/enum.go#L179-L183